CRM/Custom/Form/CustomDataByType.php

```diff
@@ -27,31 +27,39 @@
 class CRM_Custom_Form_CustomDataByType extends CRM_Core_Form {
 
   /**
-   * Contact ID associated with the Custom Data
-   *
-   * @var int
+   * @var array
    */
-  public $_contactID = NULL;
+  private $groupTree;
```

Comment:
@pradpnayak generally I try to make things private where possible because unless they are part of our fomral api they could change at any time. I have been trying to add public functions with the `@api` tag where things ARE stable to provice a reliable way to interact with forms (mostly around ids - ie ensuring forms have  a `getContactID()`.

It sounds like you have a use-case where ideally there would be a hook?

Reply:
Hi @eileenmcnaughton 

Yes, we have implemented form hooks that rely on _groupTree. It's easy to manage the array like

1. Remove certain fields from rendering on the form
2. Hide field or group set
3. Change the labels of the field based on certain conditions.

Otherwise we need to delete the element and write few code to over-ride set of Civi code.

Reply:
@pradpnayak I don't have a problem with changing it to public -  but we should probably get some issue trackign what you are trying to do so we can think about a way to ensure you are doing something supported that has unit test cover. The way in which that property is used could change at any time since it is not part of our supported api

Reply:
I thought @pradpnayak put up a PR to change this one back again?

Reply:
@eileenmcnaughton changed the visibility I think but not the name of the variable iirc

Reply:
@eileenmcnaughton https://github.com/civicrm/civicrm-core/pull/27519

Reply:
I think we should stop renaming variables during PHP8 refactors, it's risky and there's not much benefit.

Reply:
What @seamuslee001 said.

And since it's affecting hooks I'm guessing it's hard to have tests for that too, or is it?

Reply:
Ah right - yeah that one should have been renamed back too